### PR TITLE
Support Depth

### PR DESCRIPTION
**Pull request by dati91**

### CODE REVIEW

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -218,8 +218,8 @@ impl<T: RenderTarget> RenderTargetList<T> {
                 assert_eq!(t.get_render_target_layer_count(), self.targets.len());
                 assert_eq!(t.get_layer_count() as usize, self.targets.len());
                 //TODO use this
```
Comment (dati91) [inline webrender/src/tiling.rs]: Note to self: remove this line

File: webrender/src/device.rs [hunk 2]
```diff
@@ -1440,19 +1526,75 @@ impl<B: hal::Backend> Framebuffer<B> {
...
 }
 
+pub struct Depthbuffer<B: hal::Backend> {
+    pub image: B::Image,
+    pub memory: B::Memory,
+    pub image_view: B::ImageView,
+}
+
+impl<B: hal::Backend> Depthbuffer<B> {
+    pub fn new(device: &B::Device, memory_types: &[hal::MemoryType], pixel_width: u16, pixel_height: u16, depth_format: hal::format::Format) -> Self {
```
Comment (zakorgy) [inline webrender/src/device.rs]: Split this parameter list to multiple lines.

File: webrender/src/device.rs [hunk 3]
```diff
@@ -1396,10 +1473,11 @@ pub struct Framebuffer<B: hal::Backend> {
     pub layer_index: u16,
     pub image_view: B::ImageView,
     pub fbo: B::Framebuffer,
+    pub rbo: RBOId,
 }
 
 impl<B: hal::Backend> Framebuffer<B> {
-    pub fn new(device: &B::Device, texture: &Texture, image: &Image<B>, layer_index: u16, render_pass: &B::RenderPass) -> Self {
+    pub fn new(device: &B::Device, texture: &Texture, image: &Image<B>, layer_index: u16, render_pass: &(B::RenderPass, B::RenderPass), rbo: RBOId, depth: Option<&B::ImageView>) -> Self {
```
Comment (zakorgy) [inline webrender/src/device.rs]: Split this parameter list to multiple lines.

File: webrender/src/device.rs [hunk 4]
```diff
@@ -1023,54 +1033,91 @@ impl<B: hal::Backend> Program<B> {
...
+                        (PREMULTIPLIED_DEST_OUT, DepthTest::Off),
+                        (PREMULTIPLIED_DEST_OUT, LESS_EQUAL_TEST),
+                        (PREMULTIPLIED_DEST_OUT, LESS_EQUAL_WRITE),
                 ],
             };
 
-            let pipelines_descriptors = blend_states.iter().map(|blend_state| {
+            let pipelines_descriptors = pipeline_states.iter().map(|&(blend_state, depth_test)| {
+                let sp = if depth_test == DepthTest::Off {
+                    subpass
```
Comment (zakorgy) [inline webrender/src/device.rs]: Maybe we could construct the required SubPass here like:
```rust
let sp = SubPass {
    index: 0,
    main_pass: if depth_test == DepthTest::Off {&render_pass.0} else {&render_pass.1},
}
```

**Pull request merged**